Tribler/community/multichain/database.py

```diff
@@ -0,0 +1,376 @@
+        # Unpack the db_result tuples and decode the results.
+        return [str(x[0]) for x in db_result]
+
+    def contains(self, hash_requester):
+        """
+        Check if a block is existent in the persistence layer.
+        :param hash_requester: The hash_requester that is queried
+        :return: True if the block exists, else false.
+        """
+        db_query = u"SELECT hash_requester FROM multi_chain WHERE hash_requester == ? LIMIT 1"
+        db_result = self.execute(db_query, (buffer(hash_requester),)).fetchone()
+        return db_result is not None
+
+    def get_latest_sequence_number(self, public_key):
+        """
+        Return the latest sequence number known for this public_key.
+        If no block for the pk is know returns -1.
+        :param public_key: Corresponding public key
+        :return: sequence number (integer) or -1 if no block is known
+        """
+        public_key = buffer(public_key)
+        db_query = u"SELECT MAX(sequence_number) FROM (" \
+                   u"SELECT sequence_number_requester AS sequence_number " \
+                   u"FROM multi_chain WHERE public_key_requester == ? UNION " \
+                   u"SELECT sequence_number_responder AS sequence_number " \
+                   u"FROM multi_chain WHERE public_key_responder = ? )"
+        db_result = self.execute(db_query, (public_key, public_key)).fetchone()[0]
+        return db_result if db_result is not None else -1
+
+    def get_total(self, public_key):
+        """
+        Return the latest (total_up, total_down) known for this node.
+        if no block for the pk is know returns (-1,-1)
+        :param public_key: public_key of the node
+        :return: (total_up (int), total_down (int)) or (-1, -1) if no block is known.
+        """
+        public_key = buffer(public_key)
+        db_query = u"SELECT total_up, total_down, MAX(sequence_number) FROM (" \
+                   u"SELECT total_up_requester AS total_up, total_down_requester AS total_down, " \
+                   u"sequence_number_requester AS sequence_number FROM multi_chain " \
+                   u"WHERE public_key_requester == ? UNION " \
+                   u"SELECT total_up_responder AS total_up, total_down_responder AS total_down, " \
+                   u"sequence_number_responder AS sequence_number FROM multi_chain WHERE public_key_responder = ? )" \
+                   u"LIMIT 1"
+        db_result = self.execute(db_query, (public_key, public_key)).fetchone()
+        return (db_result[0], db_result[1]) if db_result[0] is not None and db_result[1] is not None \
+            else (-1, -1)
+
+    def open(self, initial_statements=True, prepare_visioning=True):
+        return super(MultiChainDB, self).open(initial_statements, prepare_visioning)
+
+    def close(self, commit=True):
+        return super(MultiChainDB, self).close(commit)
+
+    def check_database(self, database_version):
```

Comment:
This function implements the abstract check_database function, from the dispersy Database class (dispersy/database.py), of which this class is a subclass. AFAIK this function therefore should be present here. I think I would be good to do actual database conversion stuff in (Tribler/Core/Upgrade) once we need that feature.